CDTDatastore/touchdb/TD_Database.m

```diff
@@ -331,9 +331,11 @@ - (BOOL)openWithEncryptionKeyProvider:(id<CDTEncryptionKeyProvider>)provider
 
     __block BOOL result = YES;
     __weak TD_Database* weakSelf = self;
-    [_fmdbQueue inDatabase:^(FMDatabase* db) {
+    [self.fmdbQueue inTransaction:^(FMDatabase* db, BOOL* rollback) {
         TD_Database* strongSelf = weakSelf;
 
+        *rollback = YES;
```

Comment:
Why not add a comment here that the transaction will be rolled back unless the entire block completes successfully when rollback will be set to NO?
I think that will give us a nice compromise between having all the extra ` = YES` lines and being clear about what is going to happen with the transaction.